src/PluginsSystem/Tools/Microsoft.Performance.Toolkit.Plugins.Cli/Commands/Common/PackGenCommonCommand.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Extensions.Logging;
+using Microsoft.Performance.Toolkit.Plugins.Cli.Manifest;
+using Microsoft.Performance.Toolkit.Plugins.Cli.Processing;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Cli.Commands.Common
+{
+    /// <summary>
+    ///     Base class for commands that require the plugin to be processed.
+    /// </summary>
+    /// <typeparam name="TArgs">
+    ///     The type of arguments for the command.
+    /// </typeparam>
+    internal abstract class PackGenCommonCommand<TArgs>
+        : ICommand<TArgs>
+        where TArgs : PackGenCommonArgs
+    {
+        protected readonly IManifestLocatorFactory manifestLocatorFactory;
+        protected readonly IPluginArtifactsProcessor artifactsProcessor;
+        protected readonly ILogger<PackGenCommonCommand<TArgs>> logger;
+
+        protected PackGenCommonCommand(
+            IManifestLocatorFactory manifestLocatorFactory,
+            IPluginArtifactsProcessor artifactsProcessor,
+            ILogger<PackGenCommonCommand<TArgs>> logger)
+        {
+            this.manifestLocatorFactory = manifestLocatorFactory;
+            this.artifactsProcessor = artifactsProcessor;
+            this.logger = logger;
+        }
+
+        /// <inheritdoc />
+        public int Run(TArgs args)
+        {
+            if (!TryGetProcessedPluginResult(args, out ProcessedPluginResult? processedSource))
+            {
+                return 1;
```

Comment:
Will keep as is for now since we only have 0 and 1 as exit codes and it's not really used anywhere. We can add a constant file in the future when we have more codes that represent specific errors.